addon/mixins/run.js

```diff
@@ -88,18 +89,15 @@ export default Mixin.create({
       !this.isDestroyed
     );
 
-    let pendingTimers = getOrAllocate(this, '_pendingTimers', Array);
-
     let cancelId = run.later(() => {
-      let cancelIndex = pendingTimers.indexOf(cancelId);
-      pendingTimers.splice(cancelIndex, 1);
+      getOrAllocate(this, '_registeredDisposables', Map).delete(cancelId);
```

Comment:
we can't delete from something not yet allocated, we also wont try to delete something not created, so we can likely avoid this whole `getOrAllocate` during this code path, and instead interact with the map directly, as it WILL be there

Reply:
You're right in that we know the map will be there. I can probably just go ahead and remove that `getOrAllocate` call.